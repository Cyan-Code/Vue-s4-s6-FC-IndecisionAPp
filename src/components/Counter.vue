<template>
    <h1>{{ customTitle }}</h1>
    <p>{{ counter }} <sup>2</sup> = {{squareComputed}} </p>
    <p>{{ counter }}</p>
    <div>
        <button v-on:click="increse"> +1 </button>
        <button v-on:click="decrese"> -1 </button>
    </div>
</template>
<script>

export default {
    name: 'Counter',
    props: ['title'],
    data () {
        return {
            counter: 10,
        }
    },
    methods: {
        increse() {
            return this.counter++;
        },
        decrese() {
            return this.counter--;
        }
    },
    computed: {
        squareComputed() {
            return this.counter * this.counter;
        },
        customTitle () {
            return this.$props.title || 'CustomTitle'
        }
    }
}
</script>
<style>
    
</style>