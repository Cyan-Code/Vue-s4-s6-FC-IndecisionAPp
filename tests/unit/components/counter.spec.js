import { shallowMount } from "@vue/test-utils";
import Counter from "@/components/Counter";

describe("Counter Component", () => {
    let wrapper;
    beforeEach( ()=> {
        wrapper = shallowMount(Counter);
    });

    test("should de hacer match", () => {
        expect( wrapper.html ).toMatchSnapshot();
    });
    test('debe de tener el valor counter por defecto', () => { 
        expect(wrapper.find('h10').exists()).toBeFalsy();
        expect(wrapper.find('h1').text()).toBe('CustomTitle');
    });
    test('<P> debe de tener el valor por defecto de 10', () => { 
        const pTag = wrapper.findAll('p')[1];
        expect(pTag.text()).toBe('10');
    });
    test('should de incrementar y decrementar el contador', async() => { 
        const buttons = wrapper.findAll('button');
        await buttons[0].trigger('click');

        const value = wrapper.findAll('p')[1].text();
        expect(value).toBe("11");
    })
});
