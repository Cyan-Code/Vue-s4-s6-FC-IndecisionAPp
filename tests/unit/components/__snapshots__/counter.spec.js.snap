// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Component should de hacer match 1`] = `[Function]`;
